test/extensions/filters/http/ratelimit/ratelimit_integration_test.cc

```diff
@@ -506,5 +516,44 @@ TEST_P(RatelimitFilterEnvoyRatelimitedHeaderDisabledIntegrationTest,
   EXPECT_EQ(nullptr, test_server_->counter("cluster.cluster_0.ratelimit.error"));
 }
 
+TEST_P(RatelimitIntegrationTest, OverLimitAndOK) {
```

Comment:
I am assuming this tests two independent grpc clients per our discussion, but it's not super-obvious. Can you add comments?

Reply:
Actually this test 1 grpc client. This ratelimit filter looks tricky and I am not sure what the correct behavior should be. I am afraid I may break something if grpc client is shared in this case. Can we use non-cache grpc client for this filter?

Reply:
I thought the whole point is to test that two different filters sharing a cached grpc client have independent rate-limiting behavior.

@htuch lmk if you think asking for is overkill. WDYT?

Reply:
I'm not sure what the issue is here. Rate limiting is not performed on a per-connection basis; in general there is no reliable way for a filter to know it has independent channels, even if it has independent gRPC clients; perhaps for Google gRPC but definitely not for Envoy gRPC, which can still mux the same destination in the connection pool.

I think it's valid to ask for an integration test that validates two independent gRPC clients with the same destination as resulting in independent rate limiting. What you need to do is make sure the fake upstream actually models what a real rate limit server would do (which is not to rely on connection properties, but instead the rate limit descriptors).

Happy to chat further on IM or on a call.